### PR TITLE
Feature/zapier

### PR DESCRIPTION
**Pull request by acoumb**
- The dashboard is available in the _Content_ section.
- When you insert a webhook content configuration two actions will execute: content configuration gets saved into the database and webhook trigger is called (this initial invoke will be used to test the Zap trigger and get access to a sample request object).
- After a content type is added/deleted, it will be removed/added in the list of content types.

### CODE REVIEW

**Review by AndyButland — COMMENTED** [review]
Code looks good - just left a couple of minor points.

When running it locally I also just had a few comments relating to user experience:

- The "Zapier Settings" dashboard should come after the existing ones, i.e. after "Redirect URL Management".
- In the content type drop-down list, can we order them alphabetically?
- When deleting records, there's a style we have in the back-office with a confirm tick/cross implemented using a directive.  You can see this in Umbraco Forms (e.g. if you delete a field within a form).  Could we use that style here?
- If I enter something that's not a valid URL, I get a message "Zapier Content Config: Zapier configuration is invalid", which doesn't seem that relevant.  Could we  make that a bit more meaningful?
    - Same if I don't select a content type
- Clicking "Add" doesn't appear to do anything (in fact it does, but I have to refresh to see the record added).
- Could we display "Content Type Name" in the table instead of "Content Type Alias".  Editors will be more familiar with the name, and they've selected it by name from the drop-down list.

File: src/Umbraco.Cms.Integrations.Automation.Zapier/Components/NewContentPublishedComponent.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        {
+            ContentService.Published -= ContentServiceOnPublished;
+        }
+
+        private void ContentServiceOnPublished(IContentService sender, ContentPublishedEventArgs e)
+        {
+            foreach (var node in e.PublishedEntities)
+            {
+                var zapContentConfig = _zapConfigService.GetByAlias(node.ContentType.Alias);
+                if (zapContentConfig == null) return;
```
[the remark this answers is not in the document]
Comment (acoumb) [reply src/Umbraco.Cms.Integrations.Automation.Zapier/Components/NewContentPublishedComponent.cs]: Updated.

File: src/Umbraco.Cms.Integrations.Automation.Zapier/Services/ZapConfigService.cs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            }
+        }
+
+        public string Delete(int id)
+        {
+            try
+            {
+                using (var scope = _scopeProvider.CreateScope())
+                {
+                    var entity = scope.Database.Query<ZapContentConfigTable.ZapContentConfig>().First(p => p.Id == id);
```
[the remark this answers is not in the document]
Comment (acoumb) [reply src/Umbraco.Cms.Integrations.Automation.Zapier/Services/ZapConfigService.cs]: Updated

**Comment by acoumb** [comment]
Items addressed. The _Add_ operation was not completing successfully because of the call to async method in a single thread environment which was not responding, so I had to call the method by running another thread.

**Comment by AndyButland** [comment]
Changes look good.  I just pushed a small change where I think you can use async, and added some error handling in the case of a URL that doesn't respond.

Still think there's a little bit of work to do on the interface:

- If you enter a URL that doesn't respond - try http://www.test.com - it's not really clear what's going on.  There's a long pause, and though the response comes back with an error, the webhook is still added.
    - Wondering if it might be better here to just add whatever the user provides (so long as it validates as a URL).  Then add a "Test" button against each one so they can be checked after they've been added.  You can add a "spinner" to make it clearer that something is happening to the user (see e.g. [here](https://github.com/umbraco/Umbraco-Deploy/blob/v4/dev/src/Umbraco.Deploy.UI.Client/src/views/dashboards/management-dashboard.html#L5)).
- You can add more than one web hook per document type.  Maybe that's OK?  But just wanted to check you've handled that we call more than one if more than one is registered.
- Would be good to have a little bit of explanatory text at the top of the dashboard.
- If I click off to another area of the back-office, I get a "You have unsaved changes" message displayed, that's not correct (the changes are saved).  IIRC there's an attribute or directive you can add to text boxes and drop-down lists to prevent them being flagged as "dirty" when they are used (called `no-dirty-check`).

**Review by AndyButland — CHANGES_REQUESTED** [review]
Changes look good.  I just pushed a small change where I think you can use async, and added some error handling in the case of a URL that doesn't respond.

Still think there's a little bit of work to do on the interface:

- If you enter a URL that doesn't respond - try http://www.test.com - it's not really clear what's going on.  There's a long pause, and though the response comes back with an error, the webhook is still added.
    - Wondering if it might be better here to just add whatever the user provides (so long as it validates as a URL).  Then add a "Test" button against each one so they can be checked after they've been added.  You can add a "spinner" to make it clearer that something is happening to the user (see e.g. [here](https://github.com/umbraco/Umbraco-Deploy/blob/v4/dev/src/Umbraco.Deploy.UI.Client/src/views/dashboards/management-dashboard.html#L5)).
- You can add more than one web hook per document type.  Maybe that's OK?  But just wanted to check you've handled that we call more than one if more than one is registered.
- Would be good to have a little bit of explanatory text at the top of the dashboard.
- If I click off to another area of the back-office, I get a "You have unsaved changes" message displayed, that's not correct (the changes are saved).  IIRC there's an attribute or directive you can add to text boxes and drop-down lists to prevent them being flagged as "dirty" when they are used (called `no-dirty-check`).

**Comment by acoumb** [comment]
Changes implemented:

- server-side validation for content name
- URL testing from the results table
- additional description
- added dirty attribute and loading indicator.

**Review by AndyButland — APPROVED** [review]
All looks good to me now.  I just tweaked copy and layout a little bit, so please check you are happy with that before merging.

**Pull request merged**